qiskit/circuit/quantumcircuit.py

```diff
@@ -2901,27 +2924,11 @@ def parameters(self) -> ParameterView:
     @property
     def num_parameters(self) -> int:
         """The number of parameter objects in the circuit."""
-        # Avoid a (potential) object creation if we can.
-        if self._parameters is not None:
-            return len(self._parameters)
-        return len(self._unsorted_parameters())
+        return len(self._parameter_table)
 
     def _unsorted_parameters(self) -> set[Parameter]:
-        """Efficiently get all parameters in the circuit, without any sorting overhead.
-
-        .. warning::
-
-            The returned object may directly view onto the ``ParameterTable`` internals, and so
-            should not be mutated.  This is an internal performance detail.  Code outside of this
-            package should not use this method.
-        """
-        # This should be free, by accessing the actual backing data structure of the table, but that
-        # means that we need to copy it if adding keys from the global phase.
```

Comment:
Is this no longer important to warn about?

Reply:
It was a mistake, thanks - when splitting up a larger commit that had originally gone further, I'd left this removed.  Fixed in 31009c9.

(Then I realised I made a mistake in removing it in the other commit anyway - it was a different `ParameterTable` method that stopped viewing the internals mutably, not the one here!)